### DISCUSSION TITLE
Why does changing skip_dir or other config require a full resync?

### DISCUSSION THREAD

**Discussion by gusruben** [abraunegg/onedrive#1968] [Q&A] [answered]
Every once in a while, I want to include some folder to be skipped, but it's always required for me to do a `--resync` before I can change anything. I'm wondering, why is this? It doesn't seem like excluding a directory from being synced would require the re-download of every other file. I have many very small files in my OneDrive, and not very fast wifi, so it can even take days to re-download everything. Is there any way around this?

**Comment by abraunegg** [comment, marked as answer]
@gusruben 
> Every once in a while, I want to include some folder to be skipped, but it's always required for me to do a --resync before I can change anything. I'm wondering, why is this?

Because these are client side exclusions, which, the OneDrive API has zero idea about - the API assumes what it has online is what you want locally. The `--resync` is to ensure your local state is consistent with your desired configuration and files online with that desired configuration irrespective of what the OneDrive API is trying to force upon you.

> It doesn't seem like excluding a directory from being synced would require the re-download of every other file.

Files are not re-downloaded. The online state and integrity of the files locally is checked. If the file is already present locally, it is not downloaded, if it is missing, it *is* downloaded in accordance with your desired configuration.

> I have many very small files in my OneDrive, and not very fast wifi, so it can even take days to re-download everything. Is there any way around this?

Potentially this may be of use: https://github.com/abraunegg/onedrive/blob/master/docs/USAGE.md#performing-a---force-sync-without-a---resync-or-changing-your-configuration - but you may need to re-work your usage - have the directory you want to skip always excluded - then, when you want to sync it - use this option ....
